spring-integration-core/src/main/java/org/springframework/integration/gateway/GatewayProxyFactoryBean.java

```diff
@@ -452,7 +459,16 @@ private Object invokeGatewayMethod(MethodInvocation invocation, boolean runningO
 		if (paramCount == 0 && !hasPayloadExpression) {
 			if (shouldReply) {
 				if (shouldReturnMessage) {
-					return gateway.receive();
+					if (this.convertReceiveMessage) {
+						return gateway.receive();
+					}
+					else {
+						// TODO use messagingTemplate directly to avoid breaking changes
```

Comment:
See `MessagingGatewaySupport.receive()` implementation:
```
protected Object receive() {
     ...
	return this.messagingTemplate.receiveAndConvert(replyChannel, null);
}
```
So, we are not able to receive `Message` this way. Although canonical `GatewayProxyFactoryBean` logic expects to see it as a `Message`:
```
if (shouldReturnMessage) {
	return gateway.receive();
}
```
With that I assumed that original `MessagingGatewaySupport.receive()` implementation returned `Message` in ancient times. :smile: 
That's why I decided leave that method as is and use `messagingTemplate` directly.

Let me know if you'd prefer to see `MessagingGatewaySupport.receiveMessage()`, similarly with `sendAndReceiveMessage()`.